pandas/lib.pyx

```diff
@@ -1055,6 +1055,20 @@ def string_array_replace_from_nan_rep(ndarray[object, ndim=1] arr, object nan_re
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
+def object_array_decode_bytes(ndarray[object, ndim=1] arr, object encoding):
+    """Decode any instances of bytes to str in arr using the given encoding."""
```

Comment:
And by "support that" I mean transparently decode it from bytes to text for the user inside `to_csv`. I get that since it's an object dtype we can't prohibit text and bytes inside the column itself.